stream.go

```diff
@@ -45,11 +48,36 @@ Loop:
 			break Loop
 		}
 
-		reader = bufio.NewReader(stream.getResponse().Body)
+		resp := stream.getResponse()
+		if resp.StatusCode != 200 {
+			gitter.log(fmt.Sprintf("Unexpected response code %v", resp.StatusCode))
+			continue
+		}
+
+		//"The JSON stream returns messages as JSON objects that are delimited by carriage return (\r)" <- Not true crap it's (\n) only
+		reader = bufio.NewReader(resp.Body)
 		line, err := reader.ReadBytes('\n')
-		if stream.isClosed() {
+
+		//Check if the line only consists of whitespace
+		onlyWhitespace := true
+		for _, b := range line {
+			if b != ' ' && b != '\t' && b != '\r' && b != '\n' {
+				onlyWhitespace = false
+			}
+		}
+
+		if onlyWhitespace {
+			//"Parsers must be tolerant of occasional extra newline characters placed between messages."
+			currentKeepalive := time.Now().Unix() //interesting behavior of 100+ keepalives per seconds was observed
+			if currentKeepalive-lastKeepalive > 10 {
+				gitter.log("Keepalive was received")
+			}
+			continue
```

Comment:
@mrexodia I missed this and now I think it might be a problem here. two things:
1. Why should we break the routine (the loop) if we receive multiple whitespaces in less than 10 seconds?
2. It looks like `lastKeepalive` isn't updated to `currentKeepalive` and **continue** will happen in any case which stops the stream.
I am right? If yes, let's update it.